session.go

```diff
@@ -698,9 +702,20 @@ func (q *Query) defaultsFromSession() {
 	q.serialCons = s.cfg.SerialConsistency
 	q.defaultTimestamp = s.cfg.DefaultTimestamp
 	q.idempotent = s.cfg.DefaultIdempotence
+	q.metrics = make(map[string]*queryMetrics)
 	s.mu.RUnlock()
 }
 
+func (q *Query) getHostMetrics(host *HostInfo) *queryMetrics {
+	_, exists := q.metrics[host.connectAddress.String()]
```

Comment:
Only look it up once in this method?

Reply:
Considering that this call is used for both a lookup _and_ initiating a zero value, I really can't think of a more "efficient" or better way to do it, except splitting it into 2 functions, 1 to initiate the map and the other is to fetch. Personally, don't think it makes it all that better, but is that what you're thinking of?

Reply:
I meant something like this:
```go
func (q *Query) getHostMetrics(host *HostInfo) *queryMetrics {
	metrics, exists := q.metrics[host.connectAddress.String()]
	if !exists {
		// if the host is not in the map, it means it's been accessed for the first time
		metrics = &queryMetrics{attempts: 0, totalLatency: 0}
		q.metrics[host.connectAddress.String()] = metrics
	}
 	return metrics
}
```

Reply:
yea, thought about this one too, but didn't have a good opinion on it... would you think it looks better? If so, can easily change.

Reply:
I think it does look better. 🙈

Reply:
Done